<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from pocker_bot.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- Chasis defined here -->
<robot name="pocker_bot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="light_yellow">
    <color rgba="1.0 1.0 0.5 1.0"/>
  </material>
  <material name="grey">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.423529411765 0.0392156862745 1.0"/>
  </material>
  <material name="brown">
    <color rgba="0.870588235294 0.811764705882 0.764705882353 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <!-- macro for joints -->
  <!-- <macro nam="joint_wheel"> -->
  <link name="base_link">
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.0395416666667" ixy="0" ixz="0" iyy="0.106208333333" iyz="0" izz="0.106208333333"/>
    </inertial>
    <collision name="collision_chassis">
      <origin rpy="0 0 0" xyz="0 0 0.043512"/>
      <geometry>
        <cylinder length="0.054024" radius="0.08987"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.57 0 1.57" xyz="0 0 0.0165"/>
      <geometry>
        <!-- <cylinder length="0.054024" radius="0.08987"/> -->
        <mesh filename="package://pocker_bot_description/urdf/meshes/bot_base.stl" scale="0.01 0.01 0.01"/>
      </geometry>
      <material name="light_yellow"/>
      <!-- <xacro:material name="Orange"/> -->
    </visual>
    <!-- caster front -->
    <collision name="caster_front_collision">
      <origin rpy=" 0 0 0" xyz="0.0674025 0 0.0165"/>
      <geometry>
        <sphere radius="0.0165"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>0</mu>
            <mu2>0</mu2>
            <slip1>1.0</slip1>
            <slip2>1.0</slip2>
          </ode>
        </friction>
      </surface>
    </collision>
    <visual name="caster_front_visual">
      <origin rpy=" 0 0 1.57" xyz="0.0674025 0.0057 0.0005"/>
      <geometry>
        <!-- <sphere radius="0.05"/> -->
        <mesh filename="package://pocker_bot_description/urdf/meshes/castor_wheel.stl" scale="0.00015 0.00015 0.00015"/>
      </geometry>
    </visual>
    <!-- Caster front -->
    <collision name="caster_front_collision">
      <origin rpy=" 0 0 0" xyz="-0.0674025 0 0.0165"/>
      <geometry>
        <sphere radius="0.0165"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>0</mu>
            <mu2>0</mu2>
            <slip1>1.0</slip1>
            <slip2>1.0</slip2>
          </ode>
        </friction>
      </surface>
    </collision>
    <visual name="caster_front_visual">
      <origin rpy=" 0 0 -1.57" xyz="-0.0674025 -0.0057 0.0005"/>
      <geometry>
        <!-- <sphere radius="0.05"/> -->
        <mesh filename="package://pocker_bot_description/urdf/meshes/castor_wheel.stl" scale="0.00015 0.00015 0.00015"/>
      </geometry>
    </visual>
  </link>
  <link name="left_wheel_link">
    <inertial>
      <mass value="0.2"/>
      <origin rpy="0 1.5707 1.5707" xyz="0 0 0"/>
      <inertia ixx="0.000526666666667" ixy="0" ixz="0" iyy="0.000526666666667" iyz="0" izz="0.001"/>
    </inertial>
    <collision name="left_wheel_link_collision">
      <origin rpy="0 1.5707 1.5707" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.012" radius="0.043512"/>
      </geometry>
    </collision>
    <visual name="left_wheel_link_visual">
      <origin rpy="0 0 1.57" xyz="0 0 0"/>
      <geometry>
        <!-- <cylinder length="${thickness}" radius="${radius}"/> -->
        <mesh filename="package://pocker_bot_description/urdf/meshes/wheel.dae" scale="0.6 1 1"/>
      </geometry>
    </visual>
  </link>
  <joint name="joint_left_wheel_link" type="continuous">
    <origin rpy="0 0 0" xyz="0 0.10287 0.043512"/>
    <child link="left_wheel_link"/>
    <parent link="base_link"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="1.0"/>
  </joint>
  <link name="right_wheel_link">
    <inertial>
      <mass value="0.2"/>
      <origin rpy="0 1.5707 1.5707" xyz="0 0 0"/>
      <inertia ixx="0.000526666666667" ixy="0" ixz="0" iyy="0.000526666666667" iyz="0" izz="0.001"/>
    </inertial>
    <collision name="right_wheel_link_collision">
      <origin rpy="0 1.5707 1.5707" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.012" radius="0.043512"/>
      </geometry>
    </collision>
    <visual name="right_wheel_link_visual">
      <origin rpy="0 0 -1.57" xyz="0 0 0"/>
      <geometry>
        <!-- <cylinder length="${thickness}" radius="${radius}"/> -->
        <mesh filename="package://pocker_bot_description/urdf/meshes/wheel.dae" scale="0.6 1 1"/>
      </geometry>
    </visual>
  </link>
  <joint name="joint_right_wheel_link" type="continuous">
    <origin rpy="0 0 0" xyz="0 -0.10287 0.043512"/>
    <child link="right_wheel_link"/>
    <parent link="base_link"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="10000" velocity="1000"/>
    <joint_properties damping="1.0" friction="1.0"/>
  </joint>
</robot>

